contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -493,20 +496,19 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
     /**
      * @dev See {ITeleporterMessenger-getFeeInfo}
      */
-    function getFeeInfo(
-        bytes32 destinationBlockchainID,
-        uint256 messageID
-    ) external view returns (address, uint256) {
-        TeleporterFeeInfo memory feeInfo =
-            sentMessageInfo[destinationBlockchainID][messageID].feeInfo;
+    function getFeeInfo(bytes32 messageID) external view returns (address, uint256) {
+        TeleporterFeeInfo memory feeInfo = sentMessageInfo[messageID].feeInfo;
         return (feeInfo.feeTokenAddress, feeInfo.amount);
     }
 
     /**
-     * @dev Returns the next message ID to be used to send a message to the given blockchain ID.
+     * @dev Gets the next message ID to be used for a message sent from the contract instance.
+     * @return The next message ID to be used for a message sent from the contract instance.
      */
-    function getNextMessageID(bytes32 destinationBlockchainID) external view returns (uint256) {
-        return _getNextMessageID(destinationBlockchainID);
+    function getNextMessageID(bytes32 destinationBlockchainID) external view returns (bytes32) {
+        bytes32 blockchainID_ = blockchainID;
```

Comment:
what's the decision between this and calling `initializeBlockChainID`

Reply:
If we call `initializeBlockChainID`, this function could not be `view`

Reply:
I think I'm leaning towards `initializeBlockChainID`, since we'll want to intiailize it eventually, don't think having `view` is a must, and right now you can't `getNextMessageID` until it is initialized, so user's can't call it for `messageNonce=1`.

Reply:
I feel pretty strongly about keeping any `get` method as `view` actually. I think it'd be an anti-pattern to have a get method alter the state of the contract. I agree it's odd to not be able to call `getNextMessageID` before a message has been sent or received though. I think the best option here is to just make `initializeBlockchainID` public, so that it can be explicitly initialized before sending/receiving messages, if desired.

Reply:
good call, I rethought about it and it is good to have `get` methods as read only on state. Agree that we can just have `initializeBlockchainID` public.